Source/Core/Core/PowerPC/Jit64Common/Jit64AsmCommon.cpp

```diff
@@ -404,226 +404,208 @@ void CommonAsmRoutines::GenQuantizedSingleStores()
-	MOVD_xmm(XMM0, R(RSCRATCH_EXTRA));
-	if (cpu_info.bSSE4_1)
-	{
-		PMOVZXWD(XMM0, R(XMM0));
-	}
-	else
+	if (type == QUANTIZE_FLOAT)
 	{
-		PXOR(XMM1, R(XMM1));
-		PUNPCKLWD(XMM0, R(XMM1));
+		if (w) {
+			if (jit->jo.memcheck)
+			{
+				MOVD_xmm(XMM0, R(RSCRATCH_EXTRA));
+			}
+			else if (cpu_info.bSSSE3)
+			{
+				MOVD_xmm(XMM0, MRegSum(RMEM, RSCRATCH_EXTRA));
+				PSHUFB(XMM0, M(pbswapShuffle1x4));
+			}
+			else
+			{
+				LoadAndSwap(32, RSCRATCH_EXTRA, MRegSum(RMEM, RSCRATCH_EXTRA));
+				MOVD_xmm(XMM0, R(RSCRATCH_EXTRA));
+			}
+
+			UNPCKLPS(XMM0, M(m_one));
+		}
+		else
+		{
+			// FIXME? This code (in non-MMU mode) assumes all accesses are directly to RAM, i.e.
+			// don't need hardware access handling. This will definitely crash if paired loads occur
+			// from non-RAM areas, but as far as I know, this never happens. I don't know if this is
+			// for a good reason, or merely because no game does this.
+			// If we find something that actually does do this, maybe this should be changed. How
+			// much of a performance hit would it be?
+			if (jit->jo.memcheck)
+			{
+				ROL(64, R(RSCRATCH_EXTRA), Imm8(32));
+				MOVQ_xmm(XMM0, R(RSCRATCH_EXTRA));
+			}
+			else if (cpu_info.bSSSE3)
+			{
+				MOVQ_xmm(XMM0, MRegSum(RMEM, RSCRATCH_EXTRA));
+				PSHUFB(XMM0, M(pbswapShuffle2x4));
+			}
+			else
+			{
+				LoadAndSwap(64, RSCRATCH_EXTRA, MRegSum(RMEM, RSCRATCH_EXTRA));
+				ROL(64, R(RSCRATCH_EXTRA), Imm8(32));
+				MOVQ_xmm(XMM0, R(RSCRATCH_EXTRA));
+			}
+		}
+		return;
 	}
-	CVTDQ2PS(XMM0, R(XMM0));
-	SHR(32, R(RSCRATCH2), Imm8(5));
-	MOVQ_xmm(XMM1, MDisp(RSCRATCH2, (u32)(u64)m_dequantizeTableS));
-	MULPS(XMM0, R(XMM1));
-	RET();
-
-	const u8* loadPairedU16One = AlignCode4();
-	if (jit->jo.memcheck)
-		SafeLoadToReg(RSCRATCH_EXTRA, R(RSCRATCH_EXTRA), 16, 0, QUANTIZED_REGS_TO_SAVE_LOAD, false, SAFE_LOADSTORE_NO_FASTMEM | SAFE_LOADSTORE_NO_PROLOG);
-	else
-		UnsafeLoadRegToReg(RSCRATCH_EXTRA, RSCRATCH_EXTRA, 16, 0, false);
-	CVTSI2SS(XMM0, R(RSCRATCH_EXTRA));
-	SHR(32, R(RSCRATCH2), Imm8(5));
-	MULSS(XMM0, MDisp(RSCRATCH2, (u32)(u64)m_dequantizeTableS));
-	UNPCKLPS(XMM0, M(m_one));
-	RET();
 
-	const u8* loadPairedS16Two = AlignCode4();
-	if (jit->jo.memcheck)
-		SafeLoadToReg(RSCRATCH_EXTRA, R(RSCRATCH_EXTRA), 32, 0, QUANTIZED_REGS_TO_SAVE_LOAD, false, SAFE_LOADSTORE_NO_FASTMEM | SAFE_LOADSTORE_NO_PROLOG);
-	else
-		UnsafeLoadRegToReg(RSCRATCH_EXTRA, RSCRATCH_EXTRA, 32, 0, false);
-	ROL(32, R(RSCRATCH_EXTRA), Imm8(16));
-	MOVD_xmm(XMM0, R(RSCRATCH_EXTRA));
-	if (cpu_info.bSSE4_1)
+	if (w)
 	{
-		PMOVSXWD(XMM0, R(XMM0));
+		CVTSI2SS(XMM0, R(RSCRATCH_EXTRA));
+
+		if (quantize == -1)
+		{
+			SHR(32, R(RSCRATCH2), Imm8(5));
+			MULSS(XMM0, MDisp(RSCRATCH2, (u32)(u64)m_dequantizeTableS));
+		}
+		else if (quantize > 0)
+		{
+			MULSS(XMM0, M(&m_dequantizeTableS[quantize]));
+		}
+		UNPCKLPS(XMM0, M(m_one));
 	}
 	else
 	{
-		PUNPCKLWD(XMM0, R(XMM0));
-		PSRAD(XMM0, 16);
+		switch (type)
+		{
+			case QUANTIZE_U8:
```

Comment:
Everything should be one level of indentation less for switch contents. The 'c' in case should be aligned with the switch braces. e.g.

```cpp
switch (thing)
{
case label:
    // do stuff here
    break;
}
```